### PR TITLE
new version

### PR DESCRIPTION
**Pull request by Arielgordon123**
this updated should fix those issues 
#43 
#10
but more test are needed

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import migrations from './migrations';
+
+Vue.use(Vuex);
+
+const store = new Vuex.Store({
+  modules,
+  plugins: [
+    // location: AppData\Roaming\Electron\vuex.json
+    // createPersistedState(),
+    // createSharedMutations(),
```
Comment (Arielgordon123) [inline src/store/index.js]: currently the changes to the store wont be saved to file

i need to check how to solve that

Comment (baruchiro) [reply src/store/index.js]: Please see #30

Comment (Arielgordon123) [reply src/store/index.js]: Ok, switched to  `vuex-persistedstate` but not using the cookies storage.

does it matter if the data stored at local storage or cookies?

**Comment by Arielgordon123** [comment]
should resolve #30 as well

File: src/store/index.js (hunk 1 above)
Comment (baruchiro) [reply src/store/index.js]: I hope you researched the two options well.

Because they are just options, I don't know how much they fit us.
For example, is `vuex-persistedstate` support the Electron needs at all?

The answer to this question will also answer your question, because if Electron works with cookies or local storage **and save them between sessions**, you can use them.
But I don't know which one is better. We need to save a big object, like a long list of transactions. Is there a problem saving large objects on cookies or local-storage?

Comment (Arielgordon123) [reply src/store/index.js]: the [`vuex-persistedstate`](https://github.com/robinvdvleuten/vuex-persistedstate) package saves and read from the local storage by default, so it is easier to integrate it into our app.

the[`electron-store`](https://github.com/sindresorhus/electron-store) lets you store JSON Objects to file so the data will be saved to disk but then you need to update the vuex store on every change.

there is an advantage to `electron-store` like mentioned [`here`](https://github.com/sindresorhus/electron-store/issues/17#issuecomment-307561288)

there is a limitation of 10MB on local storage vs 4096 bytes on cookies.

Comment (baruchiro) [reply src/store/index.js]: Hi, let's discuss it in #30 .
Can you copy your comment to #30, and I will answer you there?

Comment (baruchiro) [reply src/store/index.js]: Hi, are you sure the file not saved with the current plugin?
Maybe it saved in another place? (Try [Everything](https://www.voidtools.com/))

I prefer to change the plugin in another PR, if we can (But don't waste time on it)

Comment (Arielgordon123) [reply src/store/index.js]: I think it is a good idea to change the plugin in separate PR because we need to think about how to get the data that already stored into the new plugin.

the current plugin should work, but there is a problem on some getter

Comment (Arielgordon123) [reply src/store/index.js]: OK, so the problem was that I've changed the name of the app by mistake.
there isn't a problem with `vuex-electron`

File: .browserslistrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+> 1%
```
Comment (baruchiro) [inline .browserslistrc]: Do you know what this file is?

File: .electron-vue/build.js [hunk 3]
```diff
@@ -1,156 +0,0 @@
...
-function clean () {
-  del.sync(['build/*', '!build/icons', '!build/icons/icon.*'])
-  console.log(`\n${doneLog}\n`)
-  process.exit()
-}
-
-function build () {
-  greeting()
-
-  setPrivateKey()
```
Comment (baruchiro) [inline .electron-vue/build.js]: Note to this comment.
I added this method to replace the KEY in the code with the **real key**, before the `electron-builder` starts.

We need to handle this or remember that and solve that next after this PR.

File: .electron-vue/build.js [hunk 4]
```diff
@@ -1,156 +0,0 @@
-'use strict'
```
Comment (baruchiro) [inline .electron-vue/build.js]: So I think this file is just for running `webpack`.
Do we **need** `webpack`?
Do we **want** `webpack`?

File: .electron-vue/dev-runner.js [hunk 5]
```diff
@@ -1,190 +0,0 @@
-'use strict'
```
Comment (baruchiro) [inline .electron-vue/dev-runner.js]: So this file is for run a dev environment.
I'm sure that you have an alternative to that. I will try to run this PR on my workspace and I will see if we have the right paths and hot-reloading and so on.

File: .browserslistrc (hunk 2 above)
Comment (Arielgordon123) [reply .browserslistrc]: this file came from the vue cli, it's let you define the browsers you support
and because we only using electron that uses chrome we really don't need this file

File: .electron-vue/dev-runner.js (hunk 5 above)
Comment (Arielgordon123) [reply .electron-vue/dev-runner.js]: there is some problem with hot reloading on files under the /src/service/

File: .eslintrc.js [hunk 6]
```diff
@@ -31,4 +31,4 @@ module.exports = {
     'no-console': process.env.NODE_ENV === 'production' ? 2 : 1,
     'linebreak-style': process.platform === "win32"? 0: 2
   }
-}
+}
```
Comment (baruchiro) [inline .eslintrc.js]: "No newline at end of file", I'm pretty sure the lint will not like it.

File: README.md [hunk 7]
```diff
@@ -70,10 +70,10 @@ Then you can run the commands below:
 yarn
 
 # serve with hot reload at localhost:9080
```
Comment (baruchiro) [inline README.md]: I even don't know what this sentence means.
But we need to check if it is still relevant.

File: README.md [hunk 8]
```diff
@@ -70,10 +70,10 @@ Then you can run the commands below:
...
-yarn dev
+yarn electron:serve
 
 # build electron application for production
-yarn build
+yarn electron:build
```
Comment (baruchiro) [inline README.md]: **Remember:** Adjust this file with the correct command.
(I adding this comment to resolve it just before the merge, so we don't forget)

File: babel.config.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  presets: ["@vue/cli-plugin-babel/preset"],
```
Comment (baruchiro) [inline babel.config.js]: I'm trusting on you.

File: package.json [hunk 10]
```diff
@@ -4,156 +4,64 @@
   "author": "Baruch Rothkoff <baruchiro@gmail.com>",
   "description": "An Electron-vue project for israeli-bank-scrapers",
   "license": "MIT",
-  "main": "./dist/electron/main.js",
+  "private": true,
```
Comment (baruchiro) [inline package.json]: OK

File: README.md (hunk 7 above)
Comment (Arielgordon123) [reply README.md]: if you cloned the repo the node_modules folder is not there so you need to run yarn in order do install all the packages that mentioned at the package.json

File: src/App.vue [hunk 11]
```diff
@@ -5,7 +5,7 @@
 </template>
 
 <script>
-import MainPage from '@/components/MainPage';
+import MainPage from './components/MainPage';
```
Comment (baruchiro) [inline src/App.vue]: I don't know which one is better.

File: src/background.js [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+  decryptProperty(importer, fields).then((decrypted) => {
+    event.reply('decryptProperty-reply', decrypted)
+  })
+});
+
+ipcMain.on('scrape', (event, installPath, scraperName, loginFields, showBrowser, logger) => {
+  scrape(installPath, scraperName, loginFields, showBrowser, onProgress).then((result) => {
+    event.reply('scrape-reply', result)
+  })
+});
```
Comment (baruchiro) [inline src/background.js]: All these methods should be in a separate file.
But I think we will do that in another PR.

File: src/background.js [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+  mainWindow = new BrowserWindow({
+    height: 563,
+    useContentSize: true,
+    width: 1000,
+    webPreferences: {
+      nodeIntegration: true,
+    },
+  });
+  
+  if (process.env.WEBPACK_DEV_SERVER_URL) {
```
Comment (baruchiro) [inline src/background.js]: But we removed the `Webpack` server, right?

File: src/components/MainPage/Importers/AddScraper.vue [hunk 14]
```diff
@@ -32,7 +32,9 @@
 
 <script>
 import { mapActions } from 'vuex';
-import { encryptProperty } from '../../../modules/encryption/credentials';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ipcRenderer } from 'electron';
+// import { encryptProperty } from "@/modules/encryption/credentials";
```
Comment (baruchiro) [inline src/components/MainPage/Importers/AddScraper.vue]: Remove comment

File: src/components/MainPage/Importers.vue [hunk 15]
```diff
@@ -69,12 +71,16 @@ export default {
...
     }),
-    ...mapGetters([
-      'scrapersWithId',
-      'importers',
-    ]),
+    ...mapGetters(['scrapersWithId', 'importers']),
+  },
+  created() {
+    ipcRenderer.send('getScrapers');
+    ipcRenderer.on('getScrapers-reply', (event, scrapers) => {
```
Comment (baruchiro) [inline src/components/MainPage/Importers.vue]: I saw in the `AddScraper.vue` you used `ipcRenderer.sendSync` and get the answer to the `const encrypted`.
Why do you use `send` and `on` here? You can't use `sendSync`?

File: src/components/MainPage/Importers/Importer.vue [hunk 16]
```diff
@@ -41,8 +42,10 @@
 <script>
 import { mapActions } from 'vuex';
 import { MessageBox } from 'element-ui';
-import scrape from '../../../modules/scrapers';
-import { decryptProperty } from '../../../modules/encryption/credentials';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ipcRenderer, remote } from 'electron';
+// import scrape from "../../../modules/scrapers";
```
Comment (baruchiro) [inline src/components/MainPage/Importers/Importer.vue]: comment

File: src/components/MainPage/Importers.vue (hunk 15 above)
Comment (baruchiro) [reply src/components/MainPage/Importers.vue]: I'm guessing it depends on whether the action in the `main` process is `async`.
This is a low-level use of `node` event loop...

File: src/components/MainPage/Importers/Importer.vue [hunk 17]
```diff
@@ -61,10 +64,17 @@ export default {
...
-      });
+    ipcRenderer.send('decryptProperty', this.importer, 'loginFields');
+    ipcRenderer.on('decryptProperty-reply', (event, decrypted) => {
+      this.decryptedImporter = decrypted;
+    });
+
+    ipcRenderer.on('onProgress-message', (event, percent, step) => {
+      this.onProgress(percent, step);
+    });
+    // decryptProperty(this.importer, "loginFields").then(decrypted => {
```
Comment (baruchiro) [inline src/components/MainPage/Importers/Importer.vue]: comment

File: src/components/MainPage/Importers/Importer.vue [hunk 18]
```diff
@@ -75,33 +85,38 @@ export default {
...
+        this.$logger.error(
+          `message: ${error.message}. Error Code:${error.code}`,
+        );
         this.$logger.verbose(error.stack);
         success = false;
         errorMessage = error.message;
       } finally {
-        this.onProgress({ percent: 1 }, 'Done!');
-        this.updateStatus(success, errorMessage);
+        // this.onProgress({ percent: 1 }, 'Done!');
```
Comment (baruchiro) [inline src/components/MainPage/Importers/Importer.vue]: comment

File: src/main.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+
+import './plugins/element.js';
+import LoggerPlugin from './plugins/logger';
+
+
+const logger = electron.remote.getGlobal('logger');
+logger.info('The renderer process got the logger');
+Vue.use(LoggerPlugin, { logger });
+
+Vue.config.productionTip = false;
```
Comment (baruchiro) [inline src/main.js]:
```suggestion
Vue.config.productionTip = process.env.NODE_ENV !== 'production';
```

File: src/router/index.js [hunk 20]
```diff
@@ -0,0 +1,21 @@
+import Vue from 'vue';
```
Comment (baruchiro) [inline src/router/index.js]: We don't need `router`, but keep it if there are no problems.

File: src/service/scrapers.service.js [hunk 21]
```diff
@@ -1,8 +1,15 @@
...
 
-export default async function scrape(installPath, scraperName, loginFields, showBrowser, logger, onProgress) {
+async function scrape(
+  installPath,
+  scraperName,
+  loginFields,
+  showBrowser,
+  onProgress,
+) {
+  console.log('onProgress :', onProgress);
```
Comment (baruchiro) [inline src/service/scrapers.service.js]: `console.log` will not print to file.

File: src/service/scrapers.service.js [hunk 22]
```diff
@@ -14,8 +21,13 @@ export default async function scrape(installPath, scraperName, loginFields, show
     // browser : Browser, // optional option from init puppeteer browser instance outside the libary scope. you can get browser diretly from puppeteer via `puppeteer.launch()` command.
     executablePath: chromePath, // string // optional. provide a patch to local chromium to be used by puppeteer. Relevant when using `israeli-bank-scrapers-core` library
   };
-  if (logger) logger.info(JSON.stringify(options));
+  // if (logger) logger.info(JSON.stringify(options));
```
Comment (baruchiro) [inline src/service/scrapers.service.js]: Comment, we need this log

File: src/store/migrations/1.js [hunk 23]
```diff
@@ -1,10 +1,10 @@
-import { randomHex } from '../../modules/encryption/crypto';
+import randomHex from '@/modules/encryption/crypto';
 
 export default {
   number: 1,
   migration(state) {
     state.Importers.importers = state.Importers.importers.map((importer) => {
-      importer.id = importer.id || randomHex();
+      importer.id = randomHex();
```
Comment (baruchiro) [inline src/store/migrations/1.js]: I'm tired. Need to think if there is a problem with replacing an existed `id` with a new one (if there is such case at all).

File: package.json [hunk 24]
```diff
@@ -4,156 +4,64 @@
...
-    "dev": "node .electron-vue/dev-runner.js",
-    "e2e": "npm run pack && mocha test/e2e",
-    "lint": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter src test",
-    "lint:fix": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter --fix src test",
-    "pack": "npm run pack:main && npm run pack:renderer",
-    "pack:main": "cross-env NODE_ENV=production webpack --progress --colors --config .electron-vue/webpack.main.config.js",
-    "pack:renderer": "cross-env NODE_ENV=production webpack --progress --colors --config .electron-vue/webpack.renderer.config.js",
+    "serve": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "electron:build": "vue-cli-service electron:build",
```
Comment (baruchiro) [inline package.json]: Where is the `vue-cli-service` configurations?

File: .electron-vue/build.js (hunk 3 above)
Comment (Arielgordon123) [reply .electron-vue/build.js]: oops, I didn't even look at this folder...
of course, we need to handle this

File: .electron-vue/build.js (hunk 4 above)
Comment (Arielgordon123) [reply .electron-vue/build.js]: Yes we need `Webpack`
Vue.js has `webpack` as a dev dependency which implies that `webpack` takes all of you `Vue.js` file and makes one large file for each file type e.g .js .html .css ...

File: src/background.js (hunk 12 above)
Comment (Arielgordon123) [reply src/background.js]: sure

File: src/background.js (hunk 13 above)
Comment (Arielgordon123) [reply src/background.js]: no

File: package.json (hunk 24 above)
Comment (Arielgordon123) [reply package.json]: [https://github.com/baruchiro/israeli-bank-scrapers-desktop/blob/a771ab23df06361d42b9752ae7e5c081d746fa31/package.json#L40-L44](https://github.com/baruchiro/israeli-bank-scrapers-desktop/blob/a771ab23df06361d42b9752ae7e5c081d746fa31/package.json#L40-L44)

File: README.md (hunk 7 above)
Comment (baruchiro) [reply README.md]: Sorry, I didn't talk about `yarn`.
I'm talking about `# serve with hot reload at localhost:9080`. But after some documentation reading, I guess it is still relevant.

File: package.json [hunk 25]
```diff
@@ -4,156 +4,64 @@
...
-    "build:web": "cross-env BUILD_TARGET=web node .electron-vue/build.js",
-    "dev": "node .electron-vue/dev-runner.js",
-    "e2e": "npm run pack && mocha test/e2e",
-    "lint": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter src test",
-    "lint:fix": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter --fix src test",
-    "pack": "npm run pack:main && npm run pack:renderer",
-    "pack:main": "cross-env NODE_ENV=production webpack --progress --colors --config .electron-vue/webpack.main.config.js",
-    "pack:renderer": "cross-env NODE_ENV=production webpack --progress --colors --config .electron-vue/webpack.renderer.config.js",
+    "serve": "vue-cli-service serve",
+    "build": "vue-cli-service build",
```
Comment (baruchiro) [inline package.json]: In the `Release` pipeline I'm using an external tool that expects for `build` script to publish with `electron-builder`.

Comment (Arielgordon123) [reply package.json]: I've changed these lines and now you can run with `yarn serve` and build with `yarn build`

File: package.json (hunk 10 above)
Comment (baruchiro) [reply package.json]: @Arielgordon123 I think we need to mark as private, to avoid publishing as package...

File: src/components/MainPage/Importers.vue [hunk 26]
```diff
@@ -69,12 +71,16 @@ export default {
...
+      this.set_scraper(scrapers);
+    });
   },
   methods: {
+    ...mapActions(['set_scraper']),
```
Comment (baruchiro) [inline src/components/MainPage/Importers.vue]: I don't understand why the `SCRAPERS` from `israeli-bank-scrapers-core` needs to imported in `scrapers.service`, then imported by `background.js`, who is expose them through the `ipc`, and then this `Importers.vue` call them to add them to the `store`, then uses them by `state` and `getters`.

File: .electron-vue/build.js (hunk 3 above)
Comment (baruchiro) [reply .electron-vue/build.js]: I will fix that with [hooks](https://www.electron.build/configuration/configuration#hooks)

Comment (baruchiro) [reply .electron-vue/build.js]: Sorry, I solved with `pre` script

File: src/components/MainPage/Importers.vue (hunk 26 above)
Comment (baruchiro) [reply src/components/MainPage/Importers.vue]: From @Arielgordon123 : There is a conflict in `ws`, when using `israeli-bank-scrapers-core` in the `renderer` process.
